experimental/GITFans/GITFans.jl

```diff
@@ -0,0 +1,604 @@
+    G = domain(homs[1])
+    Ggens = gens(G)
+    generators_new_perm = [Vector{Int}[] for x in Ggens]
+
+    for hom in homs
+      for j in 1:length(generators_new_perm)
+        img = listperm(image(hom, Ggens[j]))
+        if length(img) == 0
+          img = Int[1]
+        end
+        push!(generators_new_perm[j], img)
+      end
+    end
+
+    return generators_new_perm
+end;
+
+
+#############################################################################
+
+# user functions
+
+"""
+    is_monomial_free(I::Oscar.MPolyIdeal, vars_to_zero = [])
+
+See Prop. 3.1 in [BKR](https://arxiv.org/abs/1603.09241).
+"""
+function is_monomial_free(I::Oscar.MPolyIdeal, vars_to_zero::Vector{Int} = Int[])
+    Oscar.singular_assure(I)
+    SingI = I.gens.S
+    R = Oscar.base_ring(SingI)
+    Rgens = Oscar.gens(R)
+    nr_variables = length(Rgens)
+    poly_list = map(f -> evaluate(f, vars_to_zero, fill(R(0), length(vars_to_zero))), gens(SingI))
+
+    # perm is an n-cycle
+    perm = collect(2:nr_variables)
+    push!(perm, 1)
+
+    Rgens_permuted = Rgens[perm]
+    si = Singular.Ideal(R, poly_list)
+
+    for i in 1:nr_variables
+        if !(nr_variables in vars_to_zero)
+            si = Singular.satstd( si, Singular.MaximalIdeal(R, 1))
+            if Singular.ngens(si) == 1 && si[1] == R(1)
+                return false
+            elseif reduce(R(1), si) == 0
+                return false
+            end
+        end
+        vars_to_zero = perm[vars_to_zero]
+        permuted_gens = [evaluate(j, Rgens_permuted) for j in gens(si)]
+        si = Singular.Ideal(R, permuted_gens)
+    end
+
+    return true
+end
+
+"""
+    orbit_cones(I::Oscar.MPolyIdeal, Q::Matrix{Int}, G::Oscar.GAPGroup = symmetric_group(1))
+
+> Return orbit representatives of the group `G` on the set of those cones
+> whose defining rays are given by subsets `S` of the rows of the matrix `Q`,
+> such that the matrix `S` has full rank and such that the ideal `I` is
+> monomial-free (see [`is_monomial_free`](@ref)) w.r.t. the variables `x_i`
+> for which the `i`-th row of `Q` is not contained in `S`.
+"""
+function orbit_cones(I::Oscar.MPolyIdeal, Q::Matrix{Int}, G::Oscar.GAPGroup = symmetric_group(1))
+    nr_variables, projected_dimension = size(Q)
+
+    collector_cones = []
+
+    # We need not consider sets of smaller size because of the rank condition.
+    for k in projected_dimension:nr_variables
+        set = Hecke.subsets(nr_variables, k)
+        orbs = GAP.Globals.Orbits(G.X, GAP.GapObj(set, recursive = true), GAP.Globals.OnSets)
+#T TODO: switch to an Oscar `orbits` function
+
+        for i in [Vector{Int64}(orbs[i][1]) for i in 1:length(orbs)]
+            current_mat = Q[i,:]
+            if rank(current_mat) == projected_dimension &&
+               is_monomial_free(I, setdiff(1:nr_variables, i))
+                cone = Polymake.polytope.Cone(INPUT_RAYS = current_mat)
+                cone.FACETS
+                cone.RAYS
```

Comment:
just to  clarify, is this simply meant to trigger some computations whose results are ignored (for now)? Or what is the motivation?

Reply:
The results are only ignored at toplevel. polymake uses these results internally for checking equality of the polyhedra. The problem is that polymake creates a giant LP otherwise. In the next polymake release this is fixed and these lines can go away. I even mentioned this in the commit message to it is easy to find out what to remove once that happens.